examples/calibration/plot_compare_calibration.py

```diff
@@ -86,17 +92,19 @@ def predict_proba(self, X):
 
 from sklearn.calibration import CalibrationDisplay
 from sklearn.ensemble import RandomForestClassifier
-from sklearn.linear_model import LogisticRegression
+from sklearn.linear_model import LogisticRegressionCV
 from sklearn.naive_bayes import GaussianNB
 
 # Create classifiers
-lr = LogisticRegression()
+lr = LogisticRegressionCV(
+    Cs=np.logspace(-6, 6, 101), cv=10, scoring="neg_log_loss", max_iter=1_000
+)
 gnb = GaussianNB()
 svc = NaivelyCalibratedLinearSVC(C=1.0, dual="auto")
-rfc = RandomForestClassifier()
+rfc = RandomForestClassifier(random_state=42)
```

Comment:
As a user, it's confusing as why one model is doing a CV search and the others don't. If we need to do a search, shouldn't we be doing that for all models with their corresponding search spaces?

Reply:
I agree that ideally, all methods should have their hyperparameters tuned with a proper scoring rule.

It's very easy (and quite efficient) to do it for Logistic Regression while would require more efforts for the other methods and that would render the example much slower and verbose.

I could add an inline comment to say so.

Reply:
Also note: tuning C for the naively adapted SVC model would not fix its calibration (I tried).